client/CPlayerInterface.cpp

```diff
@@ -1709,7 +1709,8 @@ void CPlayerInterface::stopMovement()
 void CPlayerInterface::showMarketWindow(const IMarket *market, const CGHeroInstance *visitor)
 {
 	EVENT_HANDLER_CALLED_BY_CLIENT;
-	if (market->o->ID == Obj::ALTAR_OF_SACRIFICE)
+	auto * o = dynamic_cast<const CGObjectInstance *>(market);
+	if(o && o->ID == Obj::ALTAR_OF_SACRIFICE)

```

Comment:
Perhaps this object ID check should be removed and we should instead select UI based on allowed trade types?